src/main/kotlin/uk/gov/justice/digital/hmpps/prison/visitbooker/registry/exception/PrisonerValidationException.kt

```diff
@@ -1,13 +1,13 @@
 package uk.gov.justice.digital.hmpps.prison.visitbooker.registry.exception
 
 import jakarta.validation.ValidationException
-import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.enums.PrisonerValidationErrorCodes
+import uk.gov.justice.digital.hmpps.prison.visitbooker.registry.dto.enums.PrisonerValidationError
 import java.util.function.Supplier
 
-class PrisonerValidationException(val errorCodes: List<PrisonerValidationErrorCodes>) :
+class PrisonerValidationException(val error: PrisonerValidationError) :
```

Comment:
return only.1 error code if prisoner validation error instead of a list.

Reply:
Was there a reason not to give all codes? Did frontend want it that way? I think the visit-scheduler validation returns everything?

Reply:
Yes. All of these validation errors are mutually exclusive so based on @tpmcgowan's suggestion we decided to send only 1 error code back to simplify things.

Reply:
The rationale was that the set of error codes were mutually exclusive so it wouldn't make sense for there to be more than one (e.g. a prisoner can't be both released and transferred). Or, in the case of the prison itself being unsupported, that trumps everything so is the only one needed.

If more possible codes get added we could always change this and unpick them a bit more on the front-end.